curator-framework/src/main/java/org/apache/curator/framework/imps/NamespaceWatcher.java

```diff
@@ -75,4 +89,53 @@ else if ( curatorWatcher != null )
             }
         }
     }
+
+    // specialized equals()/hashCode() that makes this instance equal to the actual watcher
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if ( this == o )
+        {
+            return true;
+        }
+        if ( o == null )
+        {
+            return false;
+        }
+
+        if ( getClass() == o.getClass() )
+        {
+            NamespaceWatcher watcher = (NamespaceWatcher)o;
+
+            if ( !unfixedPath.equals(watcher.getUnfixedPath()) )
+            {
+                return false;
+            }
+
+            //noinspection SimplifiableIfStatement
+            if ( actualWatcher != null ? !actualWatcher.equals(watcher.actualWatcher) : watcher.actualWatcher != null )
+            {
+                return false;
+            }
+            return curatorWatcher != null ? curatorWatcher.equals(watcher.curatorWatcher) : watcher.curatorWatcher == null;
+        }
+
+        //noinspection SimplifiableIfStatement
+        if ( Watcher.class.isAssignableFrom(o.getClass()) )
+        {
+            return actualWatcher == o;
+        }
+
+        return false;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = actualWatcher != null ? actualWatcher.hashCode() : 0;
+        result = 31 * result + unfixedPath.hashCode();
+        result = 31 * result + (curatorWatcher != null ? curatorWatcher.hashCode() : 0);
+        return result;
```

Comment:
shouldn't this just be `return Objects.hashCode(actualWatcher)`?

If the goal is to make this hashCode() and equals() same as underlying watcher, why add in the other components to the hash?